File: appleoranges.js
    // original solution
	
	var appleCount = 0;
    var orangeCount = 0;
    
    for (i = 0; i < apple.length; i++) {
       var posA = apple[i] + a;
       if (posA >= s && posA <= t) {
           appleCount++
       }
    }
    
    for (j = 0; j < orange.length; j++) {
        var posO = orange[j] + b;
        if (posO >= s && posO <= t) {
            orangeCount++;    
        }
    }
    
    console.log(appleCount);
    console.log(orangeCount);
	
	// updated solution with es6 syntax
	
	let appleCount = 0;
    let orangeCount = 0; 
    for (let i of apple) {
       let posA = i + a;
       if (posA >= s && posA <= t) appleCount++
    }
    for (let i of orange) {
        let posO = i + b;
        if (posO >= s && posO <= t) orangeCount++;    
    }
    console.log(appleCount);
    console.log(orangeCount);

File: appleoranges.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int s = in.nextInt();
        int t = in.nextInt();
        int a = in.nextInt();
        int b = in.nextInt();
        int m = in.nextInt();
        int n = in.nextInt();
        int appleCount = 0;
        int orangeCount = 0;
        int[] apple = new int[m];
        for(int apple_i=0; apple_i < m; apple_i++){
            apple[apple_i] = in.nextInt();
            int posA = apple[apple_i] + a;
            if (posA >= s && posA <= t) appleCount++;
        }
        int[] orange = new int[n];
        for(int orange_i=0; orange_i < n; orange_i++){
            orange[orange_i] = in.nextInt();
            int posO = orange[orange_i] + b;
            if (posO >= s && posO <= t) orangeCount++;
        }
        System.out.println(appleCount);
        System.out.println(orangeCount);
    }
}
